src/server/routes/callback.js

```diff
@@ -336,7 +336,7 @@ export default async function callback(req, res) {
     let userObjectReturnedFromAuthorizeHandler
     try {
       userObjectReturnedFromAuthorizeHandler = await provider.authorize(
-        credentials
+        credentials, provider.type === "credentials" ? req : null
```

Comment:
this check is not necessary, as this code will only run for this type of providers:

https://github.com/nextauthjs/next-auth/blob/ca0f4c6fba29dee565fcbeafcb2b8c14bd4448fa/src/server/routes/callback.js#L313

Could we also maybe sanitize internal options and cookies before passing it? If we find out it was unnecessary we can loosen it up, but it may be a good idea to start stricter, I guess. I will have to check locally first if `...req` is doable, If I recall it correctly, I had problems with it before.
```suggestion
        credentials, {...req, options: {}, cookies: {}}
```